code/obj/item/storage/secure_safe.dm

```diff
@@ -398,6 +398,20 @@ TYPEINFO(/obj/item/storage/secure/ssafe)
 							src.storage.add_contents(new treasure(src))
 					else // if what we selected wasn't a valid path
 						i++ // try again
+			if (10)
+				var/list/eggs = list(/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/purple,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/mime,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/plant,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/candy,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/knight)
+
+				for (var/i=rand(1,src.storage.slots), i>0, i--)
+					var/egg = pick(eggs)
+					if (ispath(egg))
+						egg.infertile = TRUE
```

Comment:
have you tested this? I'm pretty sure this won't work because you're adding it to the path, not the new egg after it's been made